cmd/leveldbbench/leveldbbench.go

```diff
@@ -0,0 +1,622 @@
+func (i *IODistribution) GetSizeSample() uint64 {
+	randSrcMutex.Lock()
+	randFreq := randSrc.Intn(i.totalFrequency)
+	randSrcMutex.Unlock()
+
+	log.Trace().Int("randFreq", randFreq).Int("totalFreq", i.totalFrequency).Msg("Getting Size Sample")
+	var selectedRange *IORange
+	currentFreq := 0
+	for k, v := range i.ranges {
+		currentFreq += v.Frequency
+		if randFreq <= currentFreq {
+			selectedRange = &i.ranges[k]
+			break
+		}
+	}
+	if selectedRange == nil {
+		log.Fatal().Int("randFreq", randFreq).Int("totalFreq", i.totalFrequency).Msg("Potential off by 1 error in random sample")
+		return 0 // lint
+	}
+	randRange := selectedRange.EndRange - selectedRange.StartRange
+	randSrcMutex.Lock()
+	randSize := randSrc.Intn(randRange + 1)
+	randSrcMutex.Unlock()
+	return uint64(randSize + selectedRange.StartRange)
+}
+
+func parseRawSizeDistribution(dist string) (*IODistribution, error) {
+	buckets := strings.Split(dist, ",")
+	if len(buckets) == 0 {
+		return nil, fmt.Errorf("at least one size bucket must be provided")
+	}
+	ioDist := make([]IORange, 0)
+	bucketRegEx := regexp.MustCompile(`^(\d*)-(\d*):(\d*)$`)
+	for _, r := range buckets {
+		matches := bucketRegEx.FindAllStringSubmatch(r, -1)
+		if len(matches) != 1 {
+			return nil, fmt.Errorf("the bucket %s did not match expected format of start-end:ratio", r)
+		}
+		if len(matches[0]) != 4 {
+			return nil, fmt.Errorf("the bucket %s didn't match expected number of sub groups", r)
+		}
+		startRange, err := strconv.Atoi(matches[0][1])
+		if err != nil {
+			return nil, err
+		}
+		endRange, err := strconv.Atoi(matches[0][2])
+		if err != nil {
+			return nil, err
+		}
+		frequency, err := strconv.Atoi(matches[0][3])
+		if err != nil {
+			return nil, err
+		}
+		ioRange := new(IORange)
+		ioRange.StartRange = startRange
+		ioRange.EndRange = endRange
+		ioRange.Frequency = frequency
+		err = ioRange.Validate()
+		if err != nil {
+			return nil, err
+		}
+		ioDist = append(ioDist, *ioRange)
+	}
+	return NewIODistribution(ioDist)
+}
+
+func init() {
+	flagSet := LevelDBBenchCmd.PersistentFlags()
+	writeLimit = flagSet.Uint64("write-limit", 1000000, "The number of entries to write in the db")
+	readLimit = flagSet.Uint64("read-limit", 10000000, "the number of reads will attempt to complete in a given test")
+	overwriteCount = flagSet.Uint64("overwrite-count", 5, "the number of times to overwrite the data")
+	sequentialReads = flagSet.Bool("sequential-reads", false, "if true we'll perform reads sequentially")
+	sequentialWrites = flagSet.Bool("sequential-writes", false, "if true we'll perform writes in somewhat sequential manner")
+	keySize = flagSet.Uint64("key-size", 32, "The byte length of the keys that we'll use")
+	degreeOfParallelism = flagSet.Uint8("degree-of-parallelism", 2, "The number of concurrent goroutines we'll use")
+	rawSizeDistribution = flagSet.String("size-distribution", borDistribution, "the size distribution to use while testing")
```

Comment:
nice, so i can technically generate the distribution for any client and use it here

Reply:
I hope. The idea is to hard code what we found in bor but support other distributions easily.